Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugByteReporter.cpp

```diff
@@ -0,0 +1,203 @@
+    {
+        MultiplayerDebugByteReporter::AggregateBytes(byteSize);
+        m_fieldReports[fieldName].ReportBytes(byteSize);
+    }
+
+    void MultiplayerDebugComponentReporter::ReportFragmentEnd()
+    {
+        MultiplayerDebugByteReporter::ReportAggregateBytes();
+        m_componentDirtyBytes.ReportAggregateBytes();
+    }
+
+    AZStd::vector<MultiplayerDebugComponentReporter::Report> MultiplayerDebugComponentReporter::GetFieldReports()
+    {
+        AZStd::vector<Report> copy;
+        for (auto field = m_fieldReports.begin(); field != m_fieldReports.end(); ++field)
+        {
+            copy.emplace_back(field->first, &field->second);
+        }
+
+        auto sortByFrequency = [](const Report& a, const Report& b)
+        {
+            return a.second->GetTotalCount() > b.second->GetTotalCount();
+        };
+
+        AZStd::sort(copy.begin(), copy.end(), sortByFrequency);
+
+        return copy;
+    }
+
+    void MultiplayerDebugComponentReporter::Combine(const MultiplayerDebugComponentReporter& other)
+    {
+        MultiplayerDebugByteReporter::Combine(other);
+
+        for (const auto& fieldIterator : other.m_fieldReports)
+        {
+            m_fieldReports[fieldIterator.first].Combine(fieldIterator.second);
+        }
+
+        m_componentDirtyBytes.Combine(other.m_componentDirtyBytes);
+    }
+
+    void MultiplayerDebugEntityReporter::ReportField(AZ::u32 index, const char* componentName,
+        const char* fieldName, size_t byteSize)
+    {
+        if (m_currentComponentReport == nullptr)
+        {
+            std::stringstream component;
+            component << "[" << std::setw(2) << std::setfill('0') << static_cast<int>(index) << "]" << " " << componentName;
+            m_currentComponentReport = &m_componentReports[component.str().c_str()];
+        }
+
+        m_currentComponentReport->ReportField(fieldName, byteSize);
+        MultiplayerDebugByteReporter::AggregateBytes(byteSize);
+    }
+
+    void MultiplayerDebugEntityReporter::ReportFragmentEnd()
+    {
+        if (m_currentComponentReport)
+        {
+            m_currentComponentReport->ReportFragmentEnd();
+            m_currentComponentReport = nullptr;
+        }
+
+        MultiplayerDebugByteReporter::ReportAggregateBytes();
+    }
+
+    void MultiplayerDebugEntityReporter::Combine(const MultiplayerDebugEntityReporter& other)
+    {
+        MultiplayerDebugByteReporter::Combine(other);
+
+        for (const auto& componentIterator : other.m_componentReports)
+        {
+            m_componentReports[componentIterator.first].Combine(componentIterator.second);
+        }
+
+        SetEntityName(other.GetEntityName());
+    }
+
+    void MultiplayerDebugEntityReporter::Reset()
+    {
+        MultiplayerDebugByteReporter::Reset();
+
+        m_componentReports.clear();
+    }
+
+    AZStd::map<AZStd::string, MultiplayerDebugComponentReporter>& MultiplayerDebugEntityReporter::GetComponentReports()
```

Comment:
Just to check, is it possible for this to be const/const ref or is the current form intentional?

Reply:
I tried to make it a const but, alas, the code underneath won't work with const here, as it reset states under the hood during this call.